### ISSUE TITLE
Latest version breaks my NATS connection with NoConnectionExists exception

### ISSUE THREAD

**Issue by CycoPH** [danielwertheim/mynatsclient#55]
I setup the MYNatsClient like this:
```code
// Create the NATS client
var _natsClient = new NatsClient(_connectionInfo);
_natsClient.Events.OfType<ClientConnected>().Subscribe(OnConnected);
_natsClient.Events.OfType<ClientDisconnected>().Subscribe(OnDisconnected);
_natsClient.Events.OfType<ClientWorkerFailed>().Subscribe(OnWorkerFailed);
await _natsClient.SubAsync("something.to.get", OnRatesHandler);
await _natsClient.ConnectAsync();
```
-----
As of the latest version this does not work anymore.  Breaks on adding the SubAsync before the connect.
Do I now have to subscribe my handlers every time I connect to NATS?

**Comment by danielwertheim** [comment]
It should be supported. If you find time to provide a failing test or a fix or revert of the piece that is needed, then feel free to open a PR.

@watfordgnf anything you have encountered since the last release?

**Comment by watfordgnf** [comment]
No, we have not had any issues since the last release.

**Comment by danielwertheim** [comment]
Thanks @CycoPH , you were right. Bug introduced. Sorry for the inconvenience. Fix is being released in v0.16.1
